examples/kotlin-google-java/src/main/kotlin/com/connectrpc/examples/kotlin/Main.kt

```diff
@@ -33,37 +35,50 @@ class Main {
         fun main(args: Array<String>) {
             runBlocking {
                 val host = "https://demo.connectrpc.com"
+                val okHttpClient = OkHttpClient()
+                    .newBuilder()
+                    .readTimeout(Duration.ofMinutes(10))
+                    .writeTimeout(Duration.ofMinutes(10))
+                    .callTimeout(Duration.ofMinutes(10))
+                    .build()
                 val client = ProtocolClient(
-                    httpClient = ConnectOkHttpClient(
-                        OkHttpClient()
-                            .newBuilder()
-                            .readTimeout(Duration.ofMinutes(10))
-                            .writeTimeout(Duration.ofMinutes(10))
-                            .callTimeout(Duration.ofMinutes(10))
-                            .build(),
-                    ),
+                    httpClient = ConnectOkHttpClient(okHttpClient),
                     ProtocolClientConfig(
                         host = host,
                         serializationStrategy = GoogleJavaProtobufStrategy(),
                     ),
                 )
                 val elizaServiceClient = ElizaServiceClient(client)
-                connectStreaming(elizaServiceClient)
+                try {
+                    connectStreaming(elizaServiceClient)
+                } finally {
+                    okHttpClient.dispatcher.executorService.shutdown()
+                }
             }
         }
 
         private suspend fun connectStreaming(elizaServiceClient: ElizaServiceClient) {
             val stream = elizaServiceClient.converse()
             withContext(Dispatchers.IO) {
                 // Add the message the user is sending to the views.
-                stream.send(ConverseRequest.newBuilder().setSentence("hello").build())
+                stream.send(converseRequest { sentence = "hello" })
+                stream.sendClose()
                 for (streamResult in stream.resultChannel()) {
                     streamResult.maybeFold(
                         onMessage = { result ->
                             // Update the view with the response.
                             val elizaResponse = result.message
                             println(elizaResponse.sentence)
                         },
+                        onCompletion = { result ->
+                            if (result.code != Code.OK) {
+                                val connectErr = result.connectError()
+                                if (connectErr != null) {
+                                    throw connectErr
+                                }
+                                throw ConnectError(code = result.code, metadata = result.trailers)
```

Comment:
Maybe just print the error details as a response?

Reply:
I wanted to illustrate how you might want to handle errors in real code (instead of printing code/cause/trailers).

I think it also points out opportunities to make it even easier for consumers - if the Code is ever not OK, we should always have a ConnectError (even if it wraps another cause).